### PR TITLE
For #201: Bugfix: Customize Deck and Saving it Advances the Adventure

### LINKED ISSUE #201: Bug: Customize Deck and Saving it Advances the Adventure
Work Size: 1 unit (15 min)

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #201: Bugfix: Customize Deck and Saving it Advances the Adventure

- Added previous scene field to SelectNextStageSegment, which will not allow to go to next segment if the previous scene is DeckBuilderScene
- Left puzzle to define and implement other scenes that doesn't trigger stage advancement

### CODE REVIEW

**Comment by paulodamaso** [comment]
@SilasReinagel Designed a new approach, based on events: adventure advancement only triggers on `BattleWrapFinished` event

**Comment by paulodamaso** [comment]
@SilasReinagel I noticed that maybe it's better to call the advancement directly instead of handling the events. Also, `SelectNextStageSegment` methods did not make sense split, so I joined them

File: src/ProjectNeon/Assets/Features/GameProgression/ShopSegment.cs [hunk 1]
```diff
@@ -6,5 +6,5 @@ class ShopSegment : StageSegment
 {
     public override string Name => "Shop";
 
-    public override void Start() => SceneManager.LoadScene("ShopScene");
+    public override void Start() => SceneManager.LoadScene("GameScene");
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/GameProgression/ShopSegment.cs]: How can this work? How does the player ever get to the Shop, now?

**Comment by paulodamaso** [comment]
@SilasReinagel Closed due #295

**Pull request closed without merging**